### ISSUE TITLE
[BUG] Incorrect dimensions calculation in update_status?

### ISSUE THREAD

**Issue by chtenb** [sravioli/wezterm#13]
**Describe the bug**
In update-status.lua:

```
local usable_width = pane:get_dimensions().cols - tab_bar_width - 2
```

Shouldn't this use the cols of the entire window instead? The status bar is not limited to a pane, and if you have a vertical split, the pane cols are half of the actual space, which makes the right status part disappear incorrectly in some cases.

**To Reproduce**

1. Have some tabs open
2. Open a vertical split in one of the tabs
3. watch the right status disappear

**Expected behavior**
See right status as long as there is space

**Desktop (please complete the following information):**

- OS: [e.g. Windows]
- Versionwezterm 20240520-135708-b8f94c47

**Comment by sravioli** [comment]
Yes that would be better. Wezterm [exposes window dimensions](https://wezfurlong.org/wezterm/config/lua/window/get_dimensions.html) in pixels though. I do not know how to convert them into cols (the unit used)

**Comment by drzbida** [comment]
There probably is a smarter way that I can't figure out using information about the current used font and dpi (although idk if it would be more performant), but since you know the width of a pane in px and its number of cols, you can calculate the number of cols for any given width

```lua
local pane_dimensions = pane:get_dimensions()
local win_width = window:get_dimensions().pixel_width

local usable_space =
  math.floor((win_width * pane_dimensions.cols) / pane_dimensions.pixel_width)
```
